ebos/eclwriter.hh

```diff
@@ -254,33 +254,21 @@ public:
         if (reportStepNum == 0)
             return;
 
-        Scalar curTime = simulator_.time() + simulator_.timeStepSize();
-        Scalar totalCpuTime =
+        const Scalar curTime = simulator_.time() + simulator_.timeStepSize();
+        const Scalar totalCpuTime =
             simulator_.executionTimer().realTimeElapsed() +
             simulator_.setupTimer().realTimeElapsed() +
             simulator_.vanguard().externalSetupTime();
 
-        Opm::data::Wells localWellData = simulator_.problem().wellModel().wellData();
+        const auto localWellData  = simulator_.problem().wellModel().wellData();
```

Comment:
Is it necessary to copy these data? Can they be used by const reference? (In writeOutput() you need to move them out, so there I see the need.)

Reply:
> Is it necessary to copy these data?

Well, not really, but making this `const auto&` is technically "just" lifetime extension in this case.  `wellModel().wellData()` returns an object (of type `data::Wells` which is an alias of [`data::WellRates`](https://github.com/OPM/opm-common/blob/29a946eb9aefaf0f5551d1e713353a2a3de6e39b/opm/output/data/Wells.hpp#L301)) rather than a reference to one.  By making `localWellData` an actual object instead of a reference I'm relying on NRVO to construct the return value in place.

I recently learned that lifetime extension is typically implemented compilers using something close to
```C++
void* __s = __alloca(sizeof(T));  // Magic function with automatic deallocation
const T* __p = new(__s) T(std::move(return_value));
const T& x = *__p;

use(x);

__p->~T();
```
and this has made lifetime extension less appealing to me.